### ISSUE TITLE
Deprecate `accumulate_grad_batches` from Trainer constructor

### ISSUE THREAD

**Issue by daniellepintz** [PyTorchLightning/pytorch-lightning#9820]
## Proposed refactoring or deprecation

Deprecate `accumulate_grad_batches` from Trainer constructor:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3392215ef69b483bff82bf7b0270c2349b33a73c/pytorch_lightning/trainer/trainer.py#L143

in favor of user configuring [`GradientAccumulationScheduler`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/gradient_accumulation_scheduler.py)

### Motivation
We are trying to minimize the number of arguments on Trainer constructor: #9006
Right now we have two ways to configure gradient accumulation, through the Trainer constructor argument `accumulate_grad_batches`, and through the `GradientAccumulationScheduler` callback. Let's make things simpler by only having one way, and keeping with the principle of putting customization logic into callbacks. This way we can also avoid having to check if the user set both places:

https://github.com/PyTorchLightning/pytorch-lightning/blob/3392215ef69b483bff82bf7b0270c2349b33a73c/pytorch_lightning/trainer/connectors/callback_connector.py#L117-L124

This issue is similar in motivation and pitch to #9500

### Pitch

Deprecate `accumulate_grad_batches` from Trainer constructor:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3392215ef69b483bff82bf7b0270c2349b33a73c/pytorch_lightning/trainer/trainer.py#L143

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by carmocca** [comment]
The fact that this feature is implemented as a callback is a detail. It doesn't need to be and it would be better to move this responsibility to a Loop (`TrainingEpochLoop` likely).

Since we consider that this impacts the Trainer flow directly, we don't want to remove support for the flag.

This was discussed with the wider team so we are going to close this, feel welcome to continue the discussion.
